File: solutions/c/pangram/1/pangram.c
#include "pangram.h"

bool is_pangram(const char *sentence){
  if(!sentence || *sentence == '\0') return false;
  bool marks[26] = { 0 };
  short count = 0;
  while(*sentence){
    int idx = 0;
    if(*sentence >= 'A' && *sentence <= 'Z') 
      idx =  *sentence - 'A';
    else if(*sentence >= 'a' && *sentence <= 'z') 
      idx = *sentence - 'a';
    else {
      sentence++;
      continue;
    }
    if(!marks[idx]){
      count++;
      marks[idx] = true;
    }
    sentence++;
  }
  return count == 26;
}

File: solutions/csharp/pangram/1/Pangram.cs
public static class Pangram
{
    public static bool IsPangram(string word)
    {
        int count = 0;
        word = word.ToLower();
        bool[] abc = new bool[26];
        foreach(var c  in word){
            if(char.IsLetter(c)){
                if(!abc[(c - 'a')]){
                    abc[(c - 'a')] = true;
                    count++;
                }
            }
        }
        return count == 26;
    }
}
